### PR TITLE
core(fonts): handle CORS cssRules

### PR DESCRIPTION
**Pull request by patrickhulce**
After our 3.0 release a new high volume error popped up in sentry for the fonts gatherer. Turns out Chrome changed the behavior for CORS `.cssRules` property to throw instead of be `null`.

This adds a smoketest for the CORS font case and fixes that behavior.

### CODE REVIEW

**Comment by paulirish** [comment]
Thanks Chrome for starting to throw here when you didn't before. 💔

File: lighthouse-core/gather/gatherers/fonts.js [hunk 1]
```diff
@@ -124,11 +124,24 @@ function getFontFaceFromStylesheets() {
     return new Promise(resolve => {
       newNode.addEventListener('load', function onload() {
         newNode.removeEventListener('load', onload);
-        resolve(getFontFaceFromStylesheets());
+        try {
+          const stylesheet = Array.from(document.styleSheets).find(sh => sh.ownerNode === newNode);
```
Comment (brendankenny) [inline lighthouse-core/gather/gatherers/fonts.js]: I feel like `s` would be better :P

File: lighthouse-core/gather/gatherers/fonts.js [hunk 2]
```diff
@@ -138,19 +151,28 @@ function getFontFaceFromStylesheets() {
...
   for (const stylesheet of Array.from(document.styleSheets)) {
+    const cssStylesheet = /** @type {CSSStyleSheet} */ (stylesheet);
+
     try {
-      const cssStylesheet = /** @type {CSSStyleSheet} */ (stylesheet);
-      // Cross-origin stylesheets don't expose cssRules by default. We reload them w/ CORS headers.
-      if (cssStylesheet.cssRules === null && cssStylesheet.href && cssStylesheet.ownerNode &&
-        // @ts-ignore - crossOrigin exists if ownerNode is an HTMLLinkElement
-        !cssStylesheet.ownerNode.crossOrigin) {
+      // cssRules can be null or this access can throw when CORS isn't enabled, throw a matching error message.
```
Comment (brendankenny) [inline lighthouse-core/gather/gatherers/fonts.js]: a little hard to parse comment. Semicolon instead of a comma?

File: lighthouse-cli/test/fixtures/perf/fonts.html [hunk 3]
```diff
@@ -6,14 +6,14 @@
         font-family: 'Lobster';
         font-style: normal;
         font-weight: 400;
-        src: local('Lobster'), url('./lobster-v20-latin-regular.eot?#iefix') format('eot'), url('./lobster-v20-latin-regular.woff2') format('woff2');
+        src: url('./lobster-v20-latin-regular.woff2') format('woff2');
```
Comment (wardpeet) [inline lighthouse-cli/test/fixtures/perf/fonts.html]: I added the local strings just to make sure the regex in getSheetsFontFaces works. Maybe we should keep it?

File: lighthouse-cli/test/fixtures/perf/fonts.html [hunk 4]
```diff
@@ -6,14 +6,14 @@
...
       }
       @font-face {
         font-family: 'Lobster Two';
         font-style: normal;
         font-weight: 700;
         font-display: optional;
-        src: local("Lobster Two"), url("./lobster-two-v10-latin-700.woff2?delay=4000") format('woff2');
+        src: url("./lobster-two-v10-latin-700.woff2?delay=4000") format('woff2');
```
Comment (wardpeet) [inline lighthouse-cli/test/fixtures/perf/fonts.html]: same here

File: lighthouse-cli/test/fixtures/perf/fonts.html (hunk 3 above)
Comment (patrickhulce) [reply lighthouse-cli/test/fixtures/perf/fonts.html]: Ah ok, thanks for the context I'll revert :)

File: lighthouse-core/gather/gatherers/fonts.js (hunk 1 above)
Comment (patrickhulce) [reply lighthouse-core/gather/gatherers/fonts.js]: done

File: lighthouse-core/gather/gatherers/fonts.js (hunk 2 above)
Comment (patrickhulce) [reply lighthouse-core/gather/gatherers/fonts.js]: done

File: lighthouse-cli/test/fixtures/perf/fonts.html (hunk 4 above)
Comment (patrickhulce) [reply lighthouse-cli/test/fixtures/perf/fonts.html]: done
